### DISCUSSION TITLE
Serverless "Post Function" doesn't work with AWS Lambda

### DISCUSSION THREAD

**Discussion by DeeGeeMan** [Kong/kong#7263] [Ideas and feature requests]
### Summary

The **Serverless Function** plugin "post-function" plugin doesn't work with an **AWS Lambda** plugin.

### Steps To Reproduce

1. Create an **AWS Lambda** function and assign it to a service
2. Create a **Serverless Functions** post-function using the example given in the documentation
3. Send a request to your new service without the **x-custom-auth** in the header

_Note: the request is executed even without the **x-custom-auth** in the header.  However, the **pre-function** works as expected._

### Additional Details & Logs

- Kong version (`0.14.0`)

**Comment by p0pr0ck5** [comment]
This is expected behavior due to the fact that AWS Lambda short-circuits the normal proxying lifecycle. Plugins with a lower priority that AWS Lambda will not execute in the current design of Kong.

Long-term fix is likely to be supporting a true content phase handler for plugins, rather than having to return out of the access phase. Because we leverage `proxy_pass` as the content handler this is likely to involve a bit of redirection internally.
